assembl/static/js/app/views/message.js

```diff
@@ -1048,7 +1010,122 @@ var MessageView = Marionette.LayoutView.extend({
   postRender: function() {
     return;
   },
+  renderSentiments: function() {
+    var that = this;
+    var mySentiment = this.model.get('my_sentiment');
+    if(mySentiment){
+      var activeSentiment = this.model.get('my_sentiment')['@type'];
+      var activeSentimentClass = '.' + activeSentiment;
+      var test = this.$(activeSentimentClass).addClass('active');
+    }
+    var sentiment_counts = this.model.get('sentiment_counts');
+    var increment = 0;
+    var marginLeft = 5;
+    var totalCount = 0;
+    if(!_.isEmpty(sentiment_counts)){
+      $.each(sentiment_counts, function(label,count){
+        totalCount += count;
+        increment++;
+        if(increment > 1){
+          marginLeft += 10;
+        }
+        var sentimentClass = "." + label;
+        that.$(sentimentClass).show();
+        that.$(sentimentClass).css({left:marginLeft+'px'});
+      });
+      that.$('.sentiment-names-list').css({"left":(marginLeft+25)+'px'});
+      if(mySentiment){
+        if(totalCount === 1){
+          var msg1 = i18n.gettext('You');
+          
+          that.$('.sentiment-names-list > a').html(msg1);
+        }else{
+          var numOthers = totalCount - 1;
+          var msg2 = i18n.sprintf(i18n.ngettext(
+          'You and %d Another person',
+          'You and %d Other persons',
+          numOthers), numOthers);
+          
+          that.$('.sentiment-names-list > a').html(msg2);
+        }
+      }else{
+        var msg3 = i18n.sprintf(i18n.ngettext(
+        '%d person',
+        '%d persons',
+        totalCount), totalCount);
+        
+        that.$('.sentiment-names-list > a').html(msg3);
+      }
+      that.$('.js_idea-classification-region').css({"margin-top":"35px"});
+    }else{
+      that.$('.js_idea-classification-region').css({"margin-top":"0px"});
+    }
+    if(Ctx.isUserConnected()){
+      $('.emoticon').addClass('emoticon-connected');
+    }
+  },
+  onClickEmoticon:function(event){
+    var isUserConnected = Ctx.isUserConnected();
+    if(isUserConnected){
+      var currentClass = $(event.target).attr("class");
+      if(currentClass.indexOf('active') <= -1){
+        var that = this;
+        var currentUser = Ctx.getCurrentUser();
+        var clickedSentiment = $(event.target).attr("data-id");
+        var currentPost = this.model.get("@id");
+        var data = {"target": currentPost, "user": currentUser.id, "@type": clickedSentiment};
+        var payload = JSON.stringify(data);
+        $.ajax(
+          "/data/Discussion/" + Ctx.getDiscussionId() + "/posts/" + this.model.getNumericId() + "/sentiments", {
```

Comment:
Also, just like the like feature, clicking on an emotion icon should trigger a call to Piwik (trackEvent). And what does the like feature become? Is it completely replaced? Then we should migrate all likes which have been saved in the database to positive emotions, and remove all the code and model structures related to the like feature. Data and computations about likes in the get_participant_time_series_analytics API2 route should be replaced by their equivalents in emotions.